app/backend/src/couchers/servicers/donations.py

```
@@ -129,28 +130,50 @@ def Webhook(self, request, context, session):
             # amount comes in cents
             amount = int(data_object["amount"]) // 100
             receipt_url = data_object["receipt_url"]
+            payment_intent_id = data_object["payment_intent"]
+
+            # Get invoice type from charge metadata
+            metadata = data_object.get("metadata", {})
+
+            # Check if this is from Couchers (type=donation) or WooCommerce (has site_url=shop)
+            if metadata.get("type") == "donation":
+                invoice_type = InvoiceType.donation
+            elif metadata.get("site_url") == "https://shop.couchershq.org":
+                # This is from WooCommerce merch shop
+                invoice_type = InvoiceType.merch
+            else:
+                # Unknown payment source - ignore it
+                logger.warning(f"Stripe payment intent {payment_intent_id} has unknown metadata, ignoring")
+                return httpbody_pb2.HttpBody(
+                    content_type="application/json",
+                    data=json.dumps({"success": True, "ignored": True}).encode("ascii"),
+                )
```

Don't we still want to save it to the database though? Or is it enough for us to track in Stripe's UI? Maybe we should send an error to Sentry, because no one looks at the warning logs :)

---

Hmm yeah good point I should throw an error here and sentry makes sense!